scripts/mirror-builtin-actors.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# This script mirrors all releases of Filecoin's builtin-actors that have been updated in the past two years,
+# It respects GitHub API rate limits and paginates requests.
+# It performs the following operations:
+# - Downloads release assets from GitHub.
+# - Compares these assets with the existing ones in an S3 bucket.
+# - Uploads new or updated assets to the S3 bucket.
+# - Sends an alert to Slack if any uploads fail.
+
+set -eo pipefail
+
+export DEBIAN_FRONTEND=noninteractive
+
+BASE_FOLDER="$(pwd)/releases/actors"
+API_URL="https://api.github.com/repos/filecoin-project/builtin-actors/releases"
+FAILED_LOG="$(pwd)/failed_uploads.log"
+TWO_YEARS_AGO=$(date -d '3 years ago' +%s)
+
+mkdir -p "$BASE_FOLDER"
+
+# Function to extract the next page URL from GitHub API response headers for pagination.
+get_next_page_url() {
+    local headers="$1"
+    echo "$headers" | grep -oP '<\K([^>]+)(?=>; rel="next")' || echo ""
+}
+
+# Function to fetch and process releases
+fetch_and_process_releases() {
+    local api_url="$1"
+    local page_url="$api_url"
```

Comment:
What's the point of the `api_url` variable?

Reply:
used to pass an input when running the function but i will update